proto/tabletmanagerdata.proto

```diff
@@ -728,9 +739,88 @@ message CheckThrottlerResponse {
   double threshold = 3;
   // Error indicates an error retrieving the value
   string error = 4;
-  // Message 
+  // Message
   string message = 5;
   // RecentlyChecked indicates that the tablet has been hit with a user-facing check, which can then imply
   // that heartbeats lease should be renwed.
   bool recently_checked = 6;
+
+  message Metric {
+    // Name of the metric
+    string name = 1;
+    // StatusCode is HTTP compliant response code (e.g. 200 for OK)
+    int32 status_code = 2;
+    // Value is the metric value collected by the tablet
+    double value = 3;
+    // Threshold is the throttling threshold the table was comparing the value with
+    double threshold = 4;
+    // Error indicates an error retrieving the value
+    string error = 5;
+    // Message
+    string message = 6;
+    // Scope used in this check
+    string scope = 7;
+  }
+  map<string, Metric> metrics = 7;
+}
+
+message GetThrottlerStatusRequest {
+}
+
+message GetThrottlerStatusResponse {
+  // TabletAlias of probed tablet
+  string tablet_alias = 1;
+  string keyspace = 2;
+  string shard = 3;
+
+  // IsLeader indicates if the tablet is the leader of the shard, ie. the primary
+  bool is_leader = 4;
+  // IsOpen per stateManager
+  bool is_open = 5;
+  // IsEnabled per throttler configuration
+  bool is_enabled = 6;
+  // IsDormant: whether the throttler is dormant, ie has not received any checks in a while
+  // and goes into low-frequency probing mode.
+  bool is_dormant = 7;
+
+  // LagMetricQuery is the query used to check the lag metric, a constant used by the throttler.
+  string lag_metric_query = 8;
+  // CustomMetricQuery is the query used to check the custom metric, supplied by the user.
+  string custom_metric_query = 9;
```

Comment:
I guess I should be reading the docs ... but should we be able to specify multiple custom metrics queries here?

Reply:
That's a really good question and not according to current design. What I thought was to have multiple pre-defined metrics (load average, lag, disk IO utilization, concurrent processes, whatever), and one that could be set with a custom query.

However, good news:

- Internally, we do have multiple custom queries: the `custom` metric is obviously using a custom query. But also `threads_running` uses a (predefined, constant) `show global status like 'threads_running'` query. So internally not only is there nothing to prevent it, it's already in use.
- Externally, I guess we don't even need to change the interface. All we need is to allow the combination of `--metric-name` with `--custom-query`. So we can make this work. 

We'll need to figure out a way to _name_ such metrics. Right now there's a single custom metric named "custom". OK so anyway what I'm saying is that while currently we don't give the user the ability to define multiple custom queries, the current constructs allow it and it's a matter of future implementation.